Pokedex/PokemonDetailsViewController.swift

```diff
@@ -0,0 +1,141 @@
+        specialAttack.text = "Special Attack: " + String(pokemon.specialAttack)
+        specialAttack.textAlignment = .center
+        view.addSubview(specialAttack)
+        
+        specialDefense = UILabel(frame: CGRect(x: 0, y: view.frame.height * 0.82, width: view.frame.width, height: 20))
+        specialDefense.text = "Special Defense: " + String(pokemon.specialDefense)
+        specialDefense.textAlignment = .center
+        view.addSubview(specialDefense)
+        
+        species = UILabel(frame: CGRect(x: 0, y: view.frame.height * 0.85, width: view.frame.width, height: 20))
+        species.text = "Species: " + String(pokemon.species)
+        species.textAlignment = .center
+        view.addSubview(species)
+        
+        speed = UILabel(frame: CGRect(x: 0, y: view.frame.height * 0.88, width: view.frame.width, height: 20))
+        speed.text = "Speed: " + String(pokemon.speed)
+        speed.textAlignment = .center
+        view.addSubview(speed)
+        
+        total = UILabel(frame: CGRect(x: 0, y: view.frame.height * 0.91, width: view.frame.width, height: 20))
+        total.text = "Total: " + String(pokemon.total)
+        total.textAlignment = .center
+        view.addSubview(total)
+        
+        types = UILabel(frame: CGRect(x: 0, y: view.frame.height * 0.94, width: view.frame.width, height: 20))
+        types.text = "Types: " + String(describing: pokemon.types)
+        types.textAlignment = .center
+        view.addSubview(types)
+        
+        pokeball = UIButton(frame: CGRect(x: view.frame.width * 0.9, y: 70, width: view.frame.width * 0.1, height: view.frame.width * 0.1))
+        pokeball.setImage(#imageLiteral(resourceName: "bwPokeBall"), for: .normal)
+        pokeball.addTarget(self, action: #selector(addToOrDeleteFromFavorites), for: .touchUpInside)
+        view.addSubview(pokeball)
+        
+        searchWebButton = UIButton(frame: CGRect(x: view.frame.width * 0.8, y: view.frame.height * 0.9 - 5, width: view.frame.width * 0.15, height: view.frame.height * 0.075))
+        searchWebButton.setImage(#imageLiteral(resourceName: "searchWebButtonImage"), for: .normal)
+        searchWebButton.addTarget(self, action: #selector(search), for: .touchUpInside)
+        view.addSubview(searchWebButton)
+        
+        // Do any additional setup after loading the view.
+    }
+
+    func search() {
+        performSegue(withIdentifier: "toWeb", sender: self)
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        let next = segue.destination as! WebSearchViewController
+        next.poke = pokemon
+    }
+    
+    func addToOrDeleteFromFavorites() {
+        if pokeball.imageView?.image == #imageLiteral(resourceName: "bwPokeBall") {
+            pokeball.setImage(#imageLiteral(resourceName: "colorPokeBall"), for: .normal)
+            defaults.set(pokemon, forKey: "p")
```

Comment:
UserDefaults is not properly set in the App Delegate file, this results in an error at this line